### PR TITLE
feat: add mem option

### PR DESCRIPTION
**Pull request by fevac**
#### Added

- Adds an option to specify memory
- Adds memory to tnscope tn wgs

#### Changed

- Set different resources for tnscope tga and wgs

## Documentation

- [x] N/A
- [ ] Updated Balsamic documentation to reflect the changes as needed for this PR.
    - [Document Name]

### Feature Tests

- [x] Run `fleetjay` (WGS-TN case). Make sure it runs sucessfully and that log/fleetjay.sacct includes `--mem` tag for the sentieon_tnscope_wgs_tumor_normal rule

```
8304277,sbatch --account development --mail-type FAIL --mail-user XXX --qos low --dependency afterok:8304234,8304230,8304236,8304242 --error PATHTOCASES/fleetjay/logs/BALSAMIC.fleetjay.sentieon_tnscope_wgs_tumor_normal.91.sh.%j.err --output PATHTOCASES/fleetjay/logs/BALSAMIC.fleetjay.sentieon_tnscope_wgs_tumor_normal.91.sh.%j.out --profile task --acctg-freq task=10 --ntasks 36 --time 24:00:00 --mem 80000MB --partition core PATHTOCASES/fleetjay/scripts/BALSAMIC.fleetjay.sentieon_tnscope_wgs_tumor_normal.91.sh
```

### Pipeline Integrity Tests

- **Report deliver (generation of the `.hk` file)**
    - [x] N/A
    - [ ] Verified
- **TGA T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **TGA T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/N Workflow**
    - [ ] N/A
    - [x] Verified
- **QC Workflow**
    - [x] N/A
    - [ ] Verified
- **PON Workflow**
    - [x] N/A
    - [ ] Verified

### Documentation

- **Atlas documentation**
    - [x] N/A
    - [ ] Updated: [Link]
- **Web portal for Clinical Genomics**
    - [x] N/A
    - [ ] Updated: [Link]

### Panel of Normal specific criteria

- [ ] The PR includes the addition of a new Panel of Normals
- [ ] The samples have been verified to adhere to the sample selection criteria on [Atlas PoN creation instructions for Balsamic](https://atlas.scilifelab.se/infrastructure/BALSAMIC/balsamic_pon/#procedure)

### User Changes

- [x] N/A
- [ ] This PR affects the output files or results.
    - [ ] User feedback is considered unnecessary because [Justification].
    - [ ] Affected users have been included in the development process and given a chance to provide feedback.

### Infrastructure Changes

- **Stored files in Housekeeper**
    - [x] N/A
    - [ ] Updated: [Link]
- **CG (CLI and delivered/uploaded files)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Servers (configuration files on Hasta)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Scout interface**
    - [x] N/A
    - [ ] Updated: [Link]

### Validation criteria

Validation criteria to be added to validation report PR: [LINK-TO-VALIDATION-REPORT-PR from the [validations repository](https://github.com/Clinical-Genomics/validations/)]

**Version specific criteria**

- Text here or N/A

> [!IMPORTANT]
> One of the below checkboxes for validation need to be checked 

- [ ] Added version specific validation criteria to validation report
- [ ] Changes validated in standard sections: [validation-section]
- [ ] Validation criteria not necessary

## Checklist

> [!IMPORTANT]  
> Ensure that all checkboxes below are ticked before merging.

### For Developers

- **PR Description**
    - [x] Provided a comprehensive description of the PR.
    - [ ] Linked relevant user stories or issues to the PR.
- **Documentation**
    - [x] Verified and updated documentation if necessary.
- **Validation criteria**
    - [x] Completed the validation criteria section of the template.
- **Tests**
    - [x] Described and tested the functionality addressed in the PR.
    - [x] Ensured integration of the new code with existing workflows.
    - [x] Confirmed that meaningful unit tests were added for the changes introduced.
    - [x] Checked that the PR has successfully passed all relevant code smells and coverage checks.
- **Review**
    - [x] Addressed and resolved all the feedback provided during the code review process.
    - [x] Obtained final approval from designated reviewers.

### For Reviewers

- **Code**
    - [ ] Code implements the intended features or fixes the reported issue.
    - [ ] Code follows the project's coding standards and style guide.
- **Documentation**
    - [x] Pipeline changes are well-documented in the CHANGELOG and relevant documentation.
- **Validation criteria**
    - [ ] The author has completed the validation criteria section of the template
- **Tests**
    - [ ] The author provided a description of their manual testing, including consideration of edge cases and boundary
      conditions where applicable, with satisfactory results.
- **Review**
    - [ ] Confirmed that the developer has addressed all the comments during the code review.

### CODE REVIEW

File: BALSAMIC/constants/cluster_analysis.json [hunk 1]
```diff
@@ -154,7 +154,8 @@
 	},
 	"sentieon_TNscope": {
 		"time": "24:00:00",
-		"n": 36
+		"n": 36,
+		"mem": 80000
```
Comment (fevac) [inline BALSAMIC/constants/cluster_analysis.json:158]: @mathiasbio I see no memory declarations in the cluster profile for snakemake. Do we don't support this snakemake feature? I think this way is cleaner

I set it to 80G memory since that was the one used that succeeded for the failed case. But maybe is too much and makes most cases to wait for too long? Thoughts?

Comment (mathiasbio) [reply BALSAMIC/constants/cluster_analysis.json]: I'll close this outdated comment : )

File: BALSAMIC/constants/cluster_analysis.json [hunk 2]
```diff
@@ -216,6 +216,11 @@
 		"time": "24:00:00",
 		"n": 36
 	},
+	"sentieon_tnscope_wgs_tumor_normal": {
```
Comment (mathiasbio) [inline BALSAMIC/constants/cluster_analysis.json:219]: I think we should add the same for the tumor only rule probably!

Comment (fevac) [reply BALSAMIC/constants/cluster_analysis.json:219]: do we also need more mem for those? or are we good and just needs to separate resources from tga?

**Review by mathiasbio — APPROVED** [review]
Looks good! So nice that you took the extra time to add the option to specify memory. This will be a good start to optimise the workflow 🙏 for instance I'm sure there are some rules that don't require as many threads as requested due to get around memory issues

File: BALSAMIC/constants/cluster_analysis.json (hunk 2 above)
Comment (mathiasbio) [reply BALSAMIC/constants/cluster_analysis.json:219]: I'm not sure what level of mem is required for those, if maybe not needing to read the normal bam makes it use less. But maybe we can check a few TO WGS cases to see

Comment (fevac) [reply BALSAMIC/constants/cluster_analysis.json:219]: Good idea! From production, the case that used the most used 25GB max. So maybe we can leave for now.
